python/lsst/meas/algorithms/loadReferenceObjects.py

```diff
@@ -435,7 +573,67 @@ def joinMatchListWithCatalog(self, matchCat, sourceCat):
         ctrCoord = lsst.geom.SpherePoint(matchmeta.getDouble('RA'),
                                          matchmeta.getDouble('DEC'), lsst.geom.degrees)
         rad = matchmeta.getDouble('RADIUS') * lsst.geom.degrees
-        refCat = self.loadSkyCircle(ctrCoord, rad, filterName).refCat
+        try:
+            epoch = matchmeta.getDouble('EPOCH')
+        except (pexExcept.NotFoundError, pexExcept.TypeError):
+            epoch = None  # Not present, or not correct type means it's not set
+        refCat = self.loadSkyCircle(ctrCoord, rad, filterName, epoch=epoch).refCat
         refCat.sort()
         sourceCat.sort()
         return afwTable.unpackMatches(matchCat, refCat, sourceCat)
+
+    def applyProperMotions(self, catalog, epoch):
+        """Apply proper motion to the catalog
+
+        The positions in the ``catalog`` are wound to the desired
+        ``epoch`` (specified as a Modified Julian Date). The ``catalog``
+        is modified in-place.
+
+        Parameters
+        ----------
+        catalog : `lsst.afw.table.SimpleCatalog`
+            Catalog of positions, containing:
+
+            - Coordinates, retrieved by the table's coordinate key.
+            - ``coord_raErr`` : Error in Right Ascension (arcsec).
+            - ``coord_decErr`` : Error in Declination (arcsec).
+            - ``pm_ra`` : Proper motion in Right Ascension (mas/yr,
+                East positive)
+            - ``pm_raErr`` : Error in ``pm_ra`` (mas/yr), optional.
+            - ``pm_dec`` : Proper motion in Declination (mas/yr,
+                North positive)
+            - ``pm_decErr`` : Error in ``pm_dec`` (mas/yr), optional.
+            - ``epoch`` : Mean epoch of object (TAI MJD).
+
+        epoch : `float`
+            Epoch to which to move objects (TAI MJD).
+        """
+        if ("epoch" not in catalog.schema or "pm_ra" not in catalog.schema or "pm_dec" not in catalog.schema):
+            if self.config.requireProperMotion:
+                raise RuntimeError("Proper motion correction required but not available from catalog")
+            self.log.warn("Proper motion correction not available from catalog")
+            return
+        if not catalog.isContiguous():
+            raise RuntimeError("Catalog must be contiguous")
+        self.log.debug("Correcting reference catalog for proper motion to %f", epoch)
+        timeDiffsYears = (epoch - catalog["epoch"])/365.25  # Time difference, years
+        coordKey = catalog.table.getCoordKey()
+        # Compute the offset of each object due to proper motion
+        # as components of the arc of a great circle along RA and Dec
+        offsetsRaRad = catalog["pm_ra"]*timeDiffsYears
```

Comment:
I'd slightly prefer that this be done in a utility library so others can use it if needed.  Ideally it would delegate to a third party package to do the calculations, but if we don't have that, so be it.